core/store/models/webauthn.go

```diff
@@ -0,0 +1,240 @@
+	webAuthn, err := webauthn.New(&webauthn.Config{
+		RPDisplayName: "Chainlink Operator",	// Display Name
+		RPID:          config.RPID,        		// Generally the domain name
+		RPOrigin:      config.RPOrigin, 		// The origin URL for WebAuthn requests
+	})
+	
+	if err != nil {
+		return nil, err
+	}
+
+	if sessionStore == nil {
+		return nil, errors.New("SessionStore must not be nil")
+	}
+
+	waUser, err := duoWebAuthUserFromUser(user, uwas);
+	if err != nil {
+		return nil, err
+	}
+
+	registerOptions := func(credCreationOpts *protocol.PublicKeyCredentialCreationOptions) {
+		credCreationOpts.CredentialExcludeList = waUser.CredentialExcludeList()
+	}
+
+	// generate PublicKeyCredentialCreationOptions, session data
+	options, sessionData, err := webAuthn.BeginRegistration(
+		waUser,
+		registerOptions,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	userRegistrationIndexKey := fmt.Sprintf("%s-registration", user.Email)
+	err = sessionStore.SaveWebauthnSession(userRegistrationIndexKey, sessionData, ctx.Request, ctx.Writer)
+	if err != nil {
+		return nil, err
+	}
+
+	return options, nil
+}
+
+func FinishWebAuthnRegistration(user User, uwas []WebAuthn, sessionStore *session.Store, ctx *gin.Context, config WebAuthnConfiguration) (*webauthn.Credential, error) {
+	webAuthn, err := webauthn.New(&webauthn.Config{
+		RPDisplayName: "Chainlink Operator",	// Display Name
+		RPID:          config.RPID,        		// Generally the domain name
+		RPOrigin:      config.RPOrigin, 		// The origin URL for WebAuthn requests
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if sessionStore == nil {
+		return nil, errors.New("SessionStore must not be nil")
+	}
+
+	userRegistrationIndexKey := fmt.Sprintf("%s-registration", user.Email)
+	sessionData, err := sessionStore.GetWebauthnSession(userRegistrationIndexKey, ctx.Request)
+	if err != nil {
+		return nil, err
+	}
+
+	waUser, err := duoWebAuthUserFromUser(user, uwas);
+	if err != nil {
+		return nil, err
+	}
+
+	credential, err := webAuthn.FinishRegistration(waUser, sessionData, ctx.Request)
+	if err != nil {
+		logger.Errorf("Finish registration failed %v", err)
+		return nil, err
+	}
+
+	return credential, nil
+}
+
+func BeginWebAuthnLogin(user User, uwas []WebAuthn, sr SessionRequest) (*protocol.CredentialAssertion, error) {
+	webAuthn, err := webauthn.New(&webauthn.Config{
+		RPDisplayName: "Chainlink Operator",		// Display Name
+		RPID:          sr.WebAuthnConfig.RPID,      // Generally the domain name
+		RPOrigin:      sr.WebAuthnConfig.RPOrigin, 	// The origin URL for WebAuthn requests
+	})
+
+	if err != nil {
+		logger.Errorf("Could not create webAuthn structure with RPID: %s and RPOrigin: %s", sr.WebAuthnConfig.RPID, sr.WebAuthnConfig.RPOrigin)
+		return nil, err
+	}
+
+	waUser, err := duoWebAuthUserFromUser(user, uwas);
+	if err != nil {
+		logger.Errorf("Could not convert user to WebAuth user: %s", user.Email)
```

Comment:
Don't log here. If you must log, log at the top level where the error comes out.

You should Wrap here if you want extra context